cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -731,6 +733,12 @@ if(ARROW_WITH_BROTLI)
   include_directories(SYSTEM ${BROTLI_INCLUDE_DIR})
 endif()
 
+if(PARQUET_BUILD_ENCRYPTION)
```

Comment:
I think invoking `find_package` twice (again on line 743) is redundant. Why not modify the later one to QUIET and disable both `PARQUET_BUILD_ENCRYPTION` and `ARROW_WITH_GRPC` if `OPENSSL_FOUND` is false

Reply:
I think this is a question of what is the appropriate default build configuration.

* If -DARROW_PARQUET=ON, and OpenSSL is available, build with encryption, otherwise OFF
* If user opts in to building encryption (is there an option for this now? consistently named and documented?) and OpenSSL is not available, fail with error
* If user passes ARROW_FLIGHT=ON, then OpenSSL is required so not finding it should trigger an error also

Reply:
Yes, that was more my thinking. I didn't think the same behavior was acceptable for `ARROW_WITH_GRPC`.

I'll take a stab at implementing the logic that Wes outlined. That sounds right to me.